### PR TITLE
Add critical alert option to authorization request

### PR DESCRIPTION
**Pull request by franco-octavianus**
## Description

Adding the critical alert option when requesting for push notification authorization.

NOTE: This is specifically only for RMA, this changes **MIGHT** not suitable if being reused for other application.

## JIRA
https://ischemaview.atlassian.net/browse/CLOUD-4859

### CODE REVIEW

File: src/ios/PushPlugin.m [hunk 1]
```diff
@@ -219,6 +219,9 @@ - (void)init:(CDVInvokedUrlCommand*)command;
                 authorizationOptions |= UNAuthorizationOptionAlert;
             }
 
+            // Specifically for RMA, always add critical alert option on authorization request
+            authorizationOptions |= UNAuthorizationOptionCriticalAlert;
```
Comment (satyahegde) [inline src/ios/PushPlugin.m:223]: Shouldn't we add a `criticalAlertArg` at L205 and then wrap this in an if statement?

Comment (franco-octavianus) [reply src/ios/PushPlugin.m:223]: do we need to conditionally add critical alert option?
if yes, then we will need to also figure out how to update the angular (ionic) interfaces with the js interface of this plugin. (Alternatively, just use the js interface directly - which might requires further code changes to update usage of angular interface to js interface directly).

Comment (satyahegde) [reply src/ios/PushPlugin.m:223]: By conditionally, do you mean are we making `critical` an optional part of the payload? As of now, Backend is only adding `critical: 1` to the Go Notifications and not adding anything to other notifications.

In this case will this code add a value of `critical: nil` to `authorizationOptions` if BE does not provide a value for `critical`? (I'm not quite sure how this works in Objective-C, maybe there's another way to describe `undefined`) With the way this is implemented here, will backend need to add a value for the `critical` field to every notification (so `critical: 0` for non-critical alerts)?

Again, not really familiar with Objective-C, so hope that question makes sense.

Comment (franco-octavianus) [reply src/ios/PushPlugin.m:223]: Backend can still pass on the critical field in the sound payload.
This authorization options are for asking the user to give permission to send critical alert
critical alert

without the user permission, we won't be able to send critical alert

Comment (satyahegde) [reply src/ios/PushPlugin.m:223]: Ahh, I see. Okay, I think this makes sense then

**Review by satyahegde — APPROVED** [review]
LGTM (caveat being my lack of Objective-C knowledge)

**Pull request merged**